### PR TITLE
Db 1687 funding agency

### PR DESCRIPTION
**Pull request by kkuratsu**
Providing a funding agency that exists

Acceptance Criteria:

- FundingAgencyCode is an optional field, but when provided must be a valid 3-digit CGAC agency code. 
- FundingSubTierAgencyCode is an optional field, but when provided must be a valid 4-digit sub-tier agency code. 

Technical Release Notes:
New sql rules

### CODE REVIEW

File: dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+-- FundingAgencyCode is an optional field, but when provided must be a valid 3-digit CGAC agency code.
+SELECT
+    row_number,
+    funding_agency_code
+FROM detached_award_financial_assistance as dafa
+WHERE submission_id = {0}
+    AND funding_agency_code != ''
+    AND NOT EXISTS (
+        SELECT cgac_code
+        FROM cgac
```
Comment (alburde1) [inline dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql]: Stylistically, if you're using an alias on one table, you should be using that alias everywhere something from that table is referenced and you should be aliasing all other tables in the query as well for readability

File: dataactvalidator/config/sqlrules/d21_detached_award_financial_assistance.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--  FundingSubTierAgencyCode is an optional field, but when provided must be a valid 4-digit sub-tier agency code.
+SELECT
+    row_number,
+    funding_sub_tier_agency_co
+FROM detached_award_financial_assistance as dafa
+WHERE submission_id = {0}
+    AND funding_sub_tier_agency_co != ''
+    AND NOT EXISTS (
+        SELECT sub_tier_agency_code
+        FROM sub_tier_agency
```
Comment (alburde1) [inline dataactvalidator/config/sqlrules/d21_detached_award_financial_assistance.sql]: Same as above

File: tests/unit/dataactvalidator/test_d21_detached_award_financial_assistance.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ FundingSubTierAgencyCode is an optional field, but when provided
+    must be a valid 4-digit sub-tier agency code.  """
+
+    subcode = SubTierAgency(sub_tier_agency_code='0000', cgac_id='1')
+    cgac = CGAC(cgac_id='1', cgac_code='001', agency_name='test')
```
Comment (alburde1) [inline tests/unit/dataactvalidator/test_d21_detached_award_financial_assistance.py]: is cgac needed here or does the test work fine without it? I actually don't know, I'm legitimately asking.

Comment (kkuratsu) [inline tests/unit/dataactvalidator/test_d21_detached_award_financial_assistance.py]: Yes cgac is needed here. `SubTierAgency` has a relationship with `cgac`.

File: dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql (hunk 1 above)
Comment (kkuratsu) [inline dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql]: An alias is added to the table.

File: dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+-- FundingAgencyCode is an optional field, but when provided must be a valid 3-digit CGAC agency code.
+SELECT
+    row_number,
+    funding_agency_code
+FROM detached_award_financial_assistance AS dafa
```
Comment (alburde1) [inline dataactvalidator/config/sqlrules/d20_detached_award_financial_assistance.sql]: This is still not aliasing everything. In the select you should have things like `dafa.row_number` and in the WHERE all the variables should be related to an alias (like `dafa.submission_id`)

File: dataactvalidator/config/sqlrules/d21_detached_award_financial_assistance.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+--  FundingSubTierAgencyCode is an optional field, but when provided must be a valid 4-digit sub-tier agency code.
+SELECT
+    row_number,
+    funding_sub_tier_agency_co
+FROM detached_award_financial_assistance AS dafa
```
Comment (alburde1) [inline dataactvalidator/config/sqlrules/d21_detached_award_financial_assistance.sql]: Same as above.

**Pull request merged**